packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -112,6 +174,49 @@ export class CrossDomainFinalizer {
     }
   }
 
+  // Executes a confirmed L2->L1 relay transaction over the canonical bridge.
+  private async executeConfirmedL2ToL1Relay(l2TransactionHash: string, finalizationTransaction: TransactionType) {
+    try {
+      // Fetch info about the TokensBridged transaction to populate logs.
+      const l2Token = this.l2Client.getL2TokenForTokensBridgedTransactionHash(l2TransactionHash);
+      const tokensBridged = this.l2Client.getTokensBridgedForTokenBridgeTransactionHash(l2Token, l2TransactionHash);
+      const { symbol, decimals } = await this._getL2TokenInfo(l2Token);
+
+      await this.gasEstimator.update();
+      const executionResult = await runTransaction({
+        web3: this.l1Client.l1Web3,
+        transaction: finalizationTransaction,
+        transactionConfig: { ...this.gasEstimator.getCurrentFastPrice(), from: this.account },
+        availableAccounts: 1,
+        waitForMine: false,
+      });
+      if (executionResult.receipt) {
```

Comment:
When would we _not_ have a receipt? Doesn't the function above always return some non-null/undefined value as a receipt?

Reply:
that's true! removed